test/lib/cpd/CpdEngine.test.ts

```diff
@@ -52,22 +24,26 @@ describe('CpdEngine', () => {
 			const targets: RuleTarget[] = [{
 				target: '',
 				paths: ['file.cls','file.trigger','file.page','file.py','file.component']
-			}]
+			}];
 		
 			let langPathMap: Map<LANGUAGE, string[]> = new Map();
 			langPathMap.set(LANGUAGE.APEX, ['file.cls','file.trigger']);
 			langPathMap.set(LANGUAGE.VISUALFORCE, ['file.page','file.component']);
-			expect((testEngine as any).sortPaths(targets)).to.eql(langPathMap);
+			expect(testEngine.sortPaths(targets)).to.eql(langPathMap);
 		});
-	});
 
-	describe('jsonToRuleResult()', () => {
-		it('should return the correct rule result array from the json of cpd output', () => {
```

Comment:
I'd like us to have at least one test that actually checks against the specific output, and it looks like the removal of this test means there aren't any. Can we restore it, perhaps comparing the output to the contents of a file like we do for PMD?

Reply:
In PMD, the equivalent test would work on a sample file that has output interspersed with errors. None of the end-to-end tests would generate an error output and hence this test is valuable for PMD.
For CPD, we don't have any examples of errors in output xml. The sample that this test uses is a generic duplication output that all the end-to-end tests already cover. I removed it from here so that we don't feel a false sense of security with handling output interspersed with errors. When we hit a situation with errors we can recreate this test and use examples with real errors.
Hope that makes sense?

Reply:
That makes sense.